### ISSUE TITLE
Hide legacy navigation menu widget when navigation block lands in core

### ISSUE THREAD

**Issue by spacedmonkey** [WordPress/gutenberg#35781]
## What problem does this address?

When the navigation block lands in core in WP 5.9, also hide the option to create legacy navigation menu widgets. 

<img width="737" alt="Screenshot 2021-10-19 at 21 54 37" src="https://user-images.githubusercontent.com/237508/137989332-e6260875-dc84-4140-be44-97fc6a398b8d.png">

**Comment by noisysocks** [comment]
Makes sense. Noting that we can't enable the Navigation block until https://github.com/WordPress/gutenberg/issues/32952 is addressed, though, as Navigation stores its content into a Navigation entity.

Probably makes sense to tackle this and https://github.com/WordPress/gutenberg/issues/35780 at the same time.
